File: Heat/ViewModels/Plants/AddThermInfoPlantViewModel.vb
Imports System.ComponentModel.DataAnnotations

Namespace ViewModels.Plants
    Public Class AddThermInfoPlantViewModel

        Property PlantID As Integer

        <Display(name:="Classe impianto")> _
        Property PlantClassID As Integer
        Property PlantClassList As IEnumerable(Of SelectListItem)

        <Display(name:="Tipologia impianto")> _
        Property PlantTypeID As Integer
        Property PlantTypeList As IEnumerable(Of SelectListItem)


        <Display(name:="Codice dell'impianto per la provincia")> _
        Property PlantDistinctCode As String


        <Display(name:="Singola unità abitativa")> _
        Property IsSingleUnit As Boolean

        <Display(name:="Categoria energetica")> _
        Property EnergyCategory As EnergyCategoryEnum

        <Display(name:="Volume lordo riscaldato (m³)")> _
        Property GrossHeatedVolumeM3 As Single

        <Display(name:="Volume lordo raffrescato (m³)")> _
        Property GrossCooledVolumeM3 As Single
    End Class
End Namespace

File: Heat.ConvertedToC#/ViewModels/Plants/AddThermInfoPlantViewModel.cs
using System.Collections.Generic;
using System.Web.Mvc;
using System.ComponentModel.DataAnnotations;

namespace Heat.ViewModels.Plants
{
    public class AddThermInfoPlantViewModel
	{

		public int PlantID { get; set; }

		[Display(Name = "Classe impianto")]
		public int PlantClassID { get; set; }
		public IEnumerable<SelectListItem> PlantClassList { get; set; }

		[Display(Name = "Tipologia impianto")]
		public int PlantTypeID { get; set; }
		public IEnumerable<SelectListItem> PlantTypeList { get; set; }


		[Display(Name = "Codice dell'impianto per la provincia")]
		public string PlantDistinctCode { get; set; }


		[Display(Name = "Singola unità abitativa")]
		public bool IsSingleUnit { get; set; }

		[Display(Name = "Categoria energetica")]
		public EnergyCategoryEnum EnergyCategory { get; set; }

		[Display(Name = "Volume lordo riscaldato (m³)")]
		public float GrossHeatedVolumeM3 { get; set; }

		[Display(Name = "Volume lordo raffrescato (m³)")]
		public float GrossCooledVolumeM3 { get; set; }
	}
}

